src/features/events/services/events.service.ts

```diff
@@ -57,6 +62,25 @@ export class EventsService {
     );
   }
 
+  async upsert(id: number, model: VirtualEventModel){
```

Comment:
Please change the name of the endpoint from upsert to something that describes the job of the function as upsert is too vague.